scripts/identify-ids-by-query.js

```diff
@@ -0,0 +1,189 @@
+/**
+ *
+ * Given a ES query, identfies matching records and writes them to a local csv
+ *
+ * Options:
+ *  --query QUERY - Provide ES query as a quoted JSON blob
+ *  --queryfile FILE - Provide a file path to a json file with the query. Should be relative to the script and in quotes.
+ *  e.g. `--queryfile '../query.json'` in case you are in the main `discovery-hybrid-indexer` directory
+ *  --outfile FILE - Specify where to write the CSV (default ./out.csv)
+ *  --from N - Specify index to start collecting from. Default 0
+ *  --size M - Specify records per page. Default 100
+ *  --stripprefix (true|false) - Specify whether or not to strip prefix from
+ *                identifier before writing to CSV (e.g. hb12345 > 12345).
+ *                Default false
+ *  --envfile - Specify config file to use. Default ./config/qa.env
+ *
+ * Note that only one of --query and --queryfile should be used.
+ *
+ * Note that when using with `--stripprefix true`, because the output will not
+ * include nyplSource, queries should ideally restrict their scope to one
+ * nyplSource value (see example usage below):
+ *
+ * Usage:
+ *   node scripts/identify-ids-by-query --envfile [path to .env] [--outfile out.csv] --query '{"query": {
+ *      "bool": {
+ *          "must": [
+ *              {
+ *                  "regexp": {
+ *                      "idIsbn": ".*[^0-9x].*"
+ *                  }
+ *              },
+ *              {
+ *                  "term": {
+ *                      "nyplSource": "sierra-nypl"
+ *                  }
+ *              }
+ *          ]
+ *      }
+ *  }}'
+ *
+ */
+
+const dotenv = require('dotenv')
+const { parseArgs } = require('node:util')
+const fs = require('fs')
+
+const { query: esQuery, scroll: esScroll } = require('../lib/elastic-search/requests')
+const {
+  awsCredentialsFromIni,
+  castArgsToInts,
+  die
+} = require('./utils')
+const { setCredentials: kmsSetCredentials } = require('../lib/kms')
+
+/**
+* Parse script arguments from process.argv:
+**/
+const parseArguments = () => {
+  const argv = parseArgs({
+    options: {
+      envfile: {
+        type: 'string',
+        default: './config/qa.env'
+      },
+      from: {
+        type: 'string',
+        default: '0'
+      },
+      outfile: { type: 'string' },
+      query: {
+        type: 'string'
+      },
+      size: {
+        type: 'string',
+        default: '100'
+      },
+      stripprefix: {
+        type: 'boolean',
+        default: false
+      }
+    }
+  })
+
+  castArgsToInts(argv, ['batchSize', 'limit', 'offset'])
```

Comment:
I think this line shouldn't be here. These arguments are not specified in the preamble.

Reply:
Good catch. We do want to support `--limit`, so keeping that one.